geomstats/learning/em_expectation_maximization.py

```diff
@@ -0,0 +1,303 @@
+"""Expectation maximization algorithm."""
+
+import logging
+
+from sklearn.base import BaseEstimator, ClusterMixin
+
+import geomstats.backend as gs
+from geomstats.geometry.poincare_ball \
+    import PoincareBall
+from geomstats.learning._template import TransformerMixin
+from geomstats.learning.frechet_mean import FrechetMean
+
+
+EM_CONV_RATE = 1e-4
+MINIMUM_EPOCHS = 10
+DEFAULT_MAX_ITER = 100
+DEFAULT_LR = 5e-2
+DEFAULT_CONV_FACTOR = 5e-3
+DEFAULT_TOL = 1e-2
+ZETA_LOWER_BOUND = 5e-2
+ZETA_UPPER_BOUND = 2.
+ZETA_STEP = 0.001
+PDF_TOL = 1e-15
+SUM_CHECK_PDF = 1e-4
+MEAN_MAX_ITER = 150
+
+
+class RiemannianEM(TransformerMixin, ClusterMixin, BaseEstimator):
+    """Expectation-maximization class on Poincaré Ball.
+
+    A class for performing Expectation-Maximization on the
+    Poincaré Ball to fit data into a Gaussian Mixture Model.
+
+    Parameters
+    ----------
+    riemannian_metric : object of class RiemannianMetric
+        The geomstats Riemmanian metric associated with
+                        the used manifold.
+    n_gaussian : int
+        Number of Gaussian components in the mix.
+    initialisation_method : basestring
+        Choice between initialization method for variances, means and weights.
+           'random' : will select random uniformally train point as
+                     initial centroids.
+            # TODO: hzaatiti, tgeral68 implement kmeans initialisation
```

Comment:
Don't put the TODO in the docstring, otherwise it will show up on the documentation website, which will be weird for others. Put it in the code instead.